sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -171,6 +171,14 @@ object SQLConf {
       .intConf
       .createWithDefault(10)
 
+  val OPTIMIZER_INSET_SWITCH_THRESHOLD =
+    buildConf("spark.sql.optimizer.inSetSwitchThreshold")
+      .internal()
+      .doc("Configures the max set size in InSet for which Spark will generate code with " +
+        "switch statements. This is applicable only to bytes, shorts, ints, dates.")
+      .intConf
```

Comment:
To prevent user configuration errors, can we have a meaningful min/max check?
```scala
.checkValue(v => v > 0 && v < ???, ...)
```

Reply:
@kizk @mgaido91 we had a discussion about generating codes bigger than 64KB.

I am wondering if we still want to split the switch-based logic into multiple methods if we have this check suggested by @dongjoon-hyun. I've implemented the split logic locally. However, the code looks more complicated and we will need some extensions to `splitExpressionsWithCurrentInputs`.

Reply:
I am not sure why you'd need any extension. We have other parts of the code with swtich which are split. I think in general it is safer to have it.

Reply:
@mgaido91 could you point me to an example?

Reply:
ah, you're right sorry, I was remembering wrongly. There were switch based expressions for for splitting them we migrated them to a do while approach. Since the whole point of this PR is to introduce the switch construct, then I agree with you that the best way is to add a constraint here in order to have the number small enough not to cause issues with code generation.

Reply:
What about the default and max values then? The switch logic was faster than `HashSet` on 500 elements for every data type and on every machine I tested. In some cases, `HashSet` started to outperform on 550+. Also, I had to generate a set of 6000+ element to hit the limit of 64KB. My proposal is to have 400 as default and 600 as max. Then we should be safe.

Reply:
yes, sounds fine to me. Please add a comment in the codegen part in order to explain why we are not splitting the code. Thanks.

Reply:
Yeah, I'll add a comment.